compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -272,14 +272,14 @@ generateAndInstallIfaceFiles dalf src opts workDir dbPath projectPackageDatabase
             , optIsGenerated = True
             , optDflagCheck = False
             , optMbPackageName = Just unitIdStr
-            , optHideAllPkgs = True
             , optGhcCustomOpts = []
             , optPackageImports =
                   baseImports ++
                   depImps ++
                   [ exposePackage (GHC.stringToUnitId $ takeBaseName dep) True []
                   | dep <- deps
                   ]
+            , optInferDependantPackages = InferDependantPackages False
```

Comment:
Maybe add a comment here “When compiling dummy interface files for a data-dependency we know all package flags so we don’t need to infer anything.”.